### DISCUSSION TITLE
DropDown in editable DataTable row does not set selected value

### DISCUSSION THREAD

**Discussion by cjbathras** [primefaces/community#2838] [PrimeVue] [answered]
I am using an editable DataTable for displaying some settings and allowing a user to edit them. One of the columns, called Color Code, is a simple series of well-defined values so I use a DropDown as the editor for that column. In non-edit mode the value for Color Code displays correctly. However, when the user clicks the edit button for the row, the DropDown appears and no value is selected. All of the correct values are available in the DropDown, it's just the initial value is not set.

I am using PrimeVue 3.52.0 and Vue 3.4.11.

The relevant code snippets are:
```
<DataTable
  v-model:editing-rows="editingRows"
  :value="localListEntries"
  data-key="id"
  edit-mode="row"
  @row-edit-save="onRowEditSaved"
>
    <Column
      field="colorCode"
      header="Color Code"
    >
      <template #editor="{ data, field }">
        <Dropdown
          v-model="data[field]"
          :options="colorCodes"
          option-label="name"
        ></Dropdown>
      </template>
      <template #body="slotProps">
        {{ slotProps.data.colorCode }}
      </template>
    </Column>
</DataTable>

export interface LocalListEntry {
  id: number;
  colorCode?: number;
  isModified: boolean;
}
const editingRows: Ref<Array<any>> = ref([]);
const localListEntries: Ref<Array<LocalListEntry>> = ref([]);
const colorCodes = ref([
  {id: 0, name: '0'},
  {id: 1, name: '1'},
  {id: 2, name: '2'},
  {id: 3, name: '3'},
  {id: 4, name: '4'},
]);
```
The `localListEntries` array has a few valid entries in it so in my app I see a few entries in the table that I can then try to edit. Each row in non-edit mode displays correctly. In edit mode the user can change the value of Color Code and the new value sticks when save is clicked. If the user then wants to edit it again, the DropDown once again does not show the value it was changed to.

When in non-edit mode:
![image](https://github.com/user-attachments/assets/3867fe74-dd2a-4a55-a0d1-357056a2e763)
When in edit-mode:
![image](https://github.com/user-attachments/assets/82adb7bf-99a0-4ce0-b1e9-fc52382a29ed)

**Comment by cjbathras** [comment]
This has been open for a month. Anyone have any ideas for a solution?

**Reply by j0rgedev** [reply, marked as answer]
Hello @cjbathras. You forgot to set the optionValue property in the Dropdown Component. So, it would be like this:
```js
<Dropdown
    v-model="data[field]"
    :options="colorCodes"
    option-label="name"
    option-value="id"
/>
```

**Reply by cjbathras** [reply]
Oh my gosh, I can't believe I missed that. Thank you! Sometimes it just takes a second set of eyes to see exactly what's staring at us.
